salt/_modules/metalk8s_monitoring.py

```diff
@@ -0,0 +1,173 @@
+"""Utiliy methods to interact with MetalK8s monitoring.
+"""
+
+from salt.exceptions import CommandExecutionError
+
+from datetime import datetime, timedelta
+
+MISSING_DEPS = []
+
+try:
+    import requests
+    from requests.packages.urllib3.exceptions import InsecureRequestWarning
+except ImportError:
+    MISSING_DEPS.append('requests')
+
+__virtualname__ = 'metalk8s_monitoring'
+
+CONTROL_PLANE_INGRESS_PORT = 8443
+
+
+def __virtual__():
+    if MISSING_DEPS:
+        error_msg = 'Missing dependencies: {}'.format(', '.join(MISSING_DEPS))
+        return False, error_msg
+
+    requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+    return __virtualname__
+
+
+def _requests_alertmanager_api(route, method='GET', **kwargs):
+    host = __grains__['metalk8s']['control_plane_ip']
+    url = 'https://{}:{}/alertmanager/{}'.format(
+        host, CONTROL_PLANE_INGRESS_PORT, route
+    )
+
+    try:
+        response = requests.request(method, url, verify=False, **kwargs)
+    except Exception as exc:
+        raise CommandExecutionError(
+            "Unable to query Alertmanager API on {}: {}".format(url, str(exc))
+        )
+
+    try:
+        json = response.json()
+    except ValueError as exc:
+        if response.status_code != requests.codes.ok:
+            error = (
+                "Received HTTP code {} when querying Alertmanager API on {}"
+                .format(response.status_code, url)
+            )
+        else:
+            error = (
+                "Malformatted response returned from Alertmanager API: {}: {}"
```

Comment:
Mmh, I think it'd be either `Misformatted` or `Malformed`

Reply:
Changed to `Malformed`